packages/calypso-products/src/sort.js

```diff
@@ -32,23 +33,26 @@ function sortProducts( products ) {
 
 	const includedItems = products.filter( isIncludedWithPlan );
 
-	domainItems = difference( products, includedItems );
+	domainItems = products.filter( ( product ) => ! includedItems.includes( product ) );
 	domainItems = domainItems.filter( isDomainProduct );
 	domainItems = Object.entries( groupBy( domainItems, 'meta' ) );
-	domainItems = sortBy( domainItems, function ( pair ) {
+	domainItems = domainItems.sort( function ( pair ) {
 		if ( pair[ 1 ][ 0 ] && pair[ 1 ][ 0 ].cost === 0 ) {
 			return -1;
 		}
 		return pair[ 0 ];
 	} );
 	domainItems = domainItems.map( function ( pair ) {
-		return sortBy( pair[ 1 ], getDomainProductRanking );
+		return pair[ 1 ].sort( getDomainProductRanking );
 	} );
-	domainItems = flatten( domainItems );
+	domainItems = domainItems.flat();
 
 	const creditItems = products.filter( isCredits );
 
-	const otherItems = difference( products, planItems, domainItems, includedItems, creditItems );
+	const otherItems = products.filter(
+		( product ) =>
+			! [ ...planItems, ...domainItems, ...includedItems, ...creditItems ].includes( product )
+	);
```

Comment:
This version should do fewer array allocations:
```js
const allItems = [ planItems, domainItems, includedItems, creditItems ];
const otherItems = products.filter(
  ( product ) => allItems.every( ( list ) => ! list.includes( product ) )
)
```

Reply:
Thanks. That's a good optimization!